### ISSUE TITLE
edu.gcfglobal.org ZIMs are all missing some content

### ISSUE THREAD

**Issue by benoit74** [openzim/zim-requests#1052]
### ZIM(s) location

https://library.kiwix.org/#lang=&q=gcf

### Recipe(s) URL

https://farm.openzim.org/recipes?name=edu.gcfglobal.org

### Readers tested

- [ ] Kiwix-serve on iOS (iPad / iPhone)
- [ ] Kiwix-serve on Android (phone or tablet)
- [ ] Kiwix-serve on Windows
- [X] Kiwix-serve on Linux
- [ ] Kiwix-serve on Raspberry Pi (e.g. hotspot)
- [ ] Kiwix-serve on Mac
- [ ] pwa.kiwix.org
- [ ] Kiwix JS - Chrome extension
- [ ] Kiwix JS - Firefox extension
- [ ] Kiwix JS - Edge extension
- [ ] Kiwix for Android application
- [ ] Kiwix for MacOS application
- [ ] Kiwix for iOS (iPad/iPhone) application

### Which ZIM versions are impacted?

All PROD versions are impacted

### Details

Content inside courses are loaded lesson by lesson.

For instance, on https://edu.gcfglobal.org/en/beginning-a-new-career/transferring-your-skills-to-a-new-career/1/, you start with a "Continue" button at the bottom of the page. When you click on it, Lesson 2 is loaded and appears, including a new "Continue" button at the bottom of the page. And so on.

Currently, all content behind "Continue" buttons is not available in the ZIM (see https://library.kiwix.org/viewer#edu.gcfglobal.org_en_all_2024-06/edu.gcfglobal.org/en/beginning-a-new-career/how-to-decide-on-a-career-field/1/). Problem is that crawler had no idea that more content was hiding behind this "Continue" buttons.

Typical solution to solve this problem is to develop a custom behavior for Browsertrix Crawler which would fake clicks on these buttons so that the crawler fetches corresponding content.

Note for self: URL loaded by "Continue" button seems protected by a timestamp, e.g. https://edu.gcfglobal.org/en/beginning-a-new-career/transferring-your-skills-to-a-new-career/content/?_=1718454918606 ; fuzzy rule to remove this is most probably required.

@Popolechien @RavanJAltaie shall we keep the ZIM in production even if courses are incomplete? No-one complains, and Youtube video are present so it is not like we have nothing, but clearly it is incomplete. I wouldn't recommend to delete them since their limited content already provides some value + it might take some time until I develop the custom behavior.

**Comment by benoit74** [comment]
Nota: it looks like it might also be broken to properly create a ZIM of this website, even Zimit1 is failing to retrieve Youtube videos in my last tests ...

**Comment by benoit74** [comment]
UP? Do I delete production files which are significantly broken?

**Comment by Popolechien** [comment]
I'd say nobody complains because they're offline. But offering significantly broken zim files when their download is a cost is not appropriate.

**Comment by benoit74** [comment]
You're right, so I've deleted the ZIMs from the library, let's start over with these ZIMs creation and publish them once they are really ready:
- Youtube videos are working
- "Continue" buttons are working

**Comment by kelson42** [comment]
@benoit74 @rgaudin Two hours have passed and the [files are still in the library](https://library.kiwix.org/#lang=&q=gcf)! Why?

**Comment by rgaudin** [comment]
library-refresh is broken. `pokemonwiki_en_all_maxi` was moved from `other` to `zimit` without opening a ticket, as is the procedure (and a know limit).
@benoit74 is probably better informed so he'll follow-up

**Comment by outusuke** [comment]
The new ZIM file is only 500 MB, while the old one from 2024 was 6 GB, so it seems like a lot of content is missing. Also, the recipes in the ZIM file can’t be used anymore because the domain has changed to https://www.learnfree.org/
, and the old domain no longer works

**Comment by Popolechien** [comment]
I have updated the recipe to `dev`, let's see what happens with https://farm.openzim.org/pipeline/06f5d5e7-d370-434e-8304-d2987d4a09c5

**Comment by Popolechien** [comment]
Task failed.

**Comment by benoit74** [comment]
They have changed domain name.

But looking at the new website, I really doubt we can properly use Zimit on this website. This website is too complex, and relies heavily on Youtube videos which (aside from being broken currently in Zimit) are not compressed by Zimit, leading to oversized ZIMs.

**Comment by Popolechien** [comment]
@benoit74 Should we archive the recipe and delete the zim file then?

**Comment by benoit74** [comment]
Looks like a decent proposition to me.

We can probably create another issue asking for a custom scraper (seems quite "easy" based on new website) if this is a valuable resource (looks so).

**Comment by Popolechien** [comment]
Yeah but it requires login / signup - it also seems that they've changed their license to `©2026 Learnfree, Inc`

**Comment by benoit74** [comment]
Closing in favor of #1724 then
